rbc/heavydb/geo_nested_array.py

```diff
@@ -0,0 +1,336 @@
+        ptr = allocate_varlen_buffer(builder, sz, i64(8))
+        typ = sig.args[1].dtype
+        llty = context.get_value_type(typ)
+        llty_p = llty.as_pointer()
+        return builder.bitcast(ptr, llty_p)
+
+    return sig, codegen
+
+
+@extending.intrinsic
+def heavydb_geo_getitem_(typingctx, geo, index):
+    retty = typesystem.Type.fromstring(geo.item_type).tonumba()
+    sig = retty(geo, index)
+
+    def codegen(context, builder, sig, args):
+        geo, index = args
+        fnty = ir.FunctionType(void, [i8p, i64, i8p])
+        getItem = cgutils.get_or_insert_function(
+            builder.module, fnty, f"{sig.args[0].base_type}_getItem"
+        )
+        geo_ptr = builder.bitcast(
+            context.make_helper(builder, sig.args[0], value=geo)._getpointer(), i8p
+        )
+
+        # Alloca ItemType
+        fa = context.make_helper(builder, sig.return_type)
+
+        # ItemType -> void*
+        result_ptr = builder.bitcast(fa._getpointer(), i8p)
+
+        # call func
+        builder.call(getItem, [geo_ptr, index, result_ptr])
+
+        # convert void* -> ItemType
+        point_type = fa._get_be_type(fa._datamodel)
+        return builder.load(builder.bitcast(result_ptr, point_type.as_pointer()))
+
+    return sig, codegen
+
+
+@extending.overload(len)
+@extending.overload_method(GeoNestedArrayNumbaType, "size")
+def heavydb_geo_len(geo):
+    if isinstance(geo, GeoNestedArrayNumbaType):
+        base_type = geo.base_type
+        size_ = external(f"int64_t {base_type}_size(int8_t*)|cpu")
+
+        def impl(geo):
+            return size_(as_voidptr(get_alloca(geo)))
+
+        return impl
+
+
+@extending.overload_method(GeoNestedArrayNumbaType, "is_null")
+def heavydb_geo_isNull(geo):
+    if isinstance(geo, GeoNestedArrayNumbaType):
+        base_type = geo.base_type
+        isNull = external(f"bool {base_type}_isNull(int8_t*)|cpu")
+
+        def impl(geo):
+            return isNull(as_voidptr(get_alloca(geo)))
+
+        return impl
+
+
+@extending.overload(operator.getitem)
+@extending.overload_method(GeoNestedArrayNumbaType, "get_item")
+def heavydb_geo_getitem(geo, index):
+    if isinstance(geo, GeoNestedArrayNumbaType):
+
+        def impl(geo, index):
+            return heavydb_geo_getitem_(geo, index)
+
+        return impl
+
+
+def heavydb_geo_fromCoords_vec(geo, lst):
+    base_type = geo.base_type
+    fromCoords = external(
+        f"void {base_type}_fromCoords(int8_t*, double*, int64_t size)|cpu"
+    )
+
+    def impl(geo, lst):
+        sz = len(lst)
+
+        f64p = get_c_ptr(sz, nb_types.double)
+        for i in range(sz):
+            f64p[i] = lst[i]
+
+        return fromCoords(as_voidptr(get_alloca(geo)), f64p, sz)
+
+    return impl
+
+
+def heavydb_geo_fromCoords_vec2(geo, lst):
+    base_type = geo.base_type
+    fromCoords = external(
+        f"void {base_type}_fromCoords(int8_t*, double*, int64_t*, int64_t)|cpu"
+    )
+
+    def impl(geo, lst):
+        nrows = len(lst)
```

Comment:
Just a note: `n_matrices`, `nrows`, and `ncols` are confusing variable names for nested arrays that can have variable nof "columns" and that dimensionality may be greater than 2, imho.
A better set of names would be `size0`, `size1`, etc, or `dim0`, `dim1`, etc, or similar.